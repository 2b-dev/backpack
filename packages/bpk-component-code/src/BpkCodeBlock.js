import React, { PropTypes } from 'react'

import './bpk-code.scss'

const BpkCodeBlock = ({ children }) => (
  <pre className='bpk-code__pre'>
    <code className='bpk-code bpk-code--block'>{children}</code>
  </pre>
)

BpkCodeBlock.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

export default BpkCodeBlock
