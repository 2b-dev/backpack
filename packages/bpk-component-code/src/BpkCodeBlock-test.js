import React from 'react'
import renderer from 'react/lib/ReactTestRenderer'
import BpkCodeBlock from './BpkCodeBlock'

describe('BpkCodeBlock', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<BpkCodeBlock>npm install react --save-dev</BpkCodeBlock>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

