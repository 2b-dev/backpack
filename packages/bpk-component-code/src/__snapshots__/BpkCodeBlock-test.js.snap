exports[`BpkCodeBlock should render correctly 1`] = `
<pre
  className="bpk-code__pre">
  <code
    className="bpk-code bpk-code--block">
    npm install react --save-dev
  </code>
</pre>
`;
